tests/generation/test_utils.py

```diff
@@ -1236,6 +1236,69 @@ def test_prompt_lookup_decoding_matches_greedy_search(self):
             for output in (output_greedy, output_prompt_lookup):
                 self._check_outputs(output, input_ids, model.config, use_cache=True)
 
+    def test_dola_decoding_sample(self):
+        # TODO (joao): investigate skips, try to reduce incompatibilities
+        for model_class in self.all_generative_model_classes:
+            if any(model_name in model_class.__name__.lower() for model_name in ["reformer"]):
+                self.skipTest("Skip Reformer as the lm_head input size is 2 * hidden size, adopted from Rev Nets.")
+            if any(
+                model_name in model_class.__name__.lower()
+                for model_name in [
+                    "wav2vec",
+                    "clvp",
+                    "bark",
+                ]
+            ):
+                self.skipTest("Skip speech models")
```

Comment:
I previously skipped these speech models because they don't have the regular output_embeddings to perform early exit. It's actually not just because they are speech models. 

Thus, I changed this part to 

```
if model.get_output_embeddings() is None:
    self.skipTest("DoLa is not supported for models that don't have output embeddings")
```